truffle/test/VaultNotaryTest.js

```diff
@@ -0,0 +1,384 @@
+        const vaultId = uuidToHex(uuidv4(), true);
+        await truffleAssert.reverts(contract.grantUpdateHashPermission(vaultId, CARRIER, {from: ATTACKER}));
+    });
+
+
+    //*******************partially testing revokeUpdateHashPermission************************
+    it("should revert the revokeUpdateHashPermission transaction if vaultId is not registered", async () => {
+        //first create a vault
+        const vaultId = uuidToHex(uuidv4(), true);
+        await truffleAssert.reverts(contract.revokeUpdateHashPermission(vaultId, CARRIER, {from: SHIPPER}));
+    });
+
+    it("should revert the revokeUpdateHashPermission not called from the vaultOwner", async () => {
+        //first create a vault
+        const vaultId = uuidToHex(uuidv4(), true);
+        await truffleAssert.reverts(contract.revokeUpdateHashPermission(vaultId, CARRIER, {from: ATTACKER}));
+    });
+
+    //****************partially testing grantUpdateUriPermission***********************
+
+    it("should revert the grantUpdateUriPermission transaction if vaultId is not registered", async () => {
+        //first create a vault
+        const vaultId = uuidToHex(uuidv4(), true);
+        await truffleAssert.reverts(contract.grantUpdateUriPermission(vaultId, CARRIER, {from: SHIPPER}));
+    });
+
+    it("should revert the grantUpdateUriPermission, if not called from the vaultOwner", async () => {
+        //first create a vault
+        const vaultId = uuidToHex(uuidv4(), true);
+        await truffleAssert.reverts(contract.grantUpdateUriPermission(vaultId, CARRIER, {from: ATTACKER}));
+    });
+
+
+    //*******************partially testing revokeUpdateUriPermission************************
+    it("should revert the revokeUpdateUriPermission transaction if vaultId is not registered", async () => {
+        //first create a vault
+        const vaultId = uuidToHex(uuidv4(), true);
+        await truffleAssert.reverts(contract.revokeUpdateUriPermission(vaultId, CARRIER, {from: SHIPPER}));
+    });
+
+    it("should revert the revokeUpdateUriPermission not called from the vaultOwner", async () => {
+        //first create a vault
+        const vaultId = uuidToHex(uuidv4(), true);
+        await truffleAssert.reverts(contract.revokeUpdateUriPermission(vaultId, CARRIER, {from: ATTACKER}));
+    });
+
+
+    //**********Uri: testing setVaultUri and grantUpdateUriPermission and revokeUpdateUriPermission********
+
+    it("should set the uri, if the address is the SHIPPER, after registerVault", async () => {
+        const vaultId = await registerVault();
+        const setUriTx = await contract.setVaultUri(vaultId, "new_uri",  {from: SHIPPER});
+        await truffleAssert.eventEmitted(setUriTx, "VaultUri", ev => {
+            return ev.vaultUri === "new_uri" && ev.vaultId === uuidToHex32(vaultId) && ev.msgSender === SHIPPER;
+        });
+
+        const data = await contract.getVaultNotaryDetails(vaultId);
+        assert.equal(data.vaultUri, "new_uri");
```

Comment:
I would also prefer to use constants for all URI/hash values/assertions - reduces the chance for typos and makes it easier to change if necessary